server/eclipse-project/src/main/java/us/freeandfair/corla/model/CountyDashboard.java

```diff
@@ -735,82 +731,109 @@ public void setBallotsInManifest(final Integer the_ballots_in_manifest) {
    * @param the_map The map.
    * @param the_key The key.
    */
-  private void ensureCounterExists(final Map<AuditReason, Integer> the_map,
-                                   final AuditReason the_key) {
+  private void ensureCounterExists(final Map<AuditSelection, Integer> the_map,
+                                   final AuditSelection the_key) {
     if (!the_map.containsKey(the_key)) {
       the_map.put(the_key, 0);
     }
   }
   
   /**
    * @return the numbers of discrepancies found in the audit so far, 
-   * categorized by contest audit reason.
+   * categorized by contest audit selection.
    */
-  public Map<AuditReason, Integer> discrepancies() {
+  public Map<AuditSelection, Integer> discrepancies() {
     return Collections.unmodifiableMap(my_discrepancies);
   }
   
   /**
-   * Adds a discrepancy for the specified audit reason. This adds it both to the 
+   * Adds a discrepancy for the specified audit reasons. This adds it both to the 
    * total and to the current audit round, if one is ongoing.
    * 
-   * @param the_reason The reason.
+   * @param the_reasons The reasons.
    */
-  public void addDiscrepancy(final AuditReason the_reason) {
-    ensureCounterExists(my_discrepancies, the_reason);
-    my_discrepancies.put(the_reason, my_discrepancies.get(the_reason) + 1); 
+  public void addDiscrepancy(final Set<AuditReason> the_reasons) {
+    final Set<AuditSelection> selections = new HashSet<>();
+    for (final AuditReason r : the_reasons) {
+      selections.add(r.selection());
+    }
+    for (final AuditSelection s : selections) {
+      ensureCounterExists(my_discrepancies, s);
+      my_discrepancies.put(s, my_discrepancies.get(s) + 1);
+    }
     if (my_current_round_index != null) {
-      my_rounds.get(my_current_round_index).addDiscrepancy(the_reason);
+      my_rounds.get(my_current_round_index).addDiscrepancy(the_reasons);
     } 
   }
   
   /**
-   * Removes a discrepancy for the specified audit reason. This removes it 
+   * Removes a discrepancy for the specified audit reasons. This removes it 
    * both from the total and from the current audit round, if one is ongoing.
    * 
-   * @param the_reason The reason.
+   * 
+   * @param the_reasons The reasons.
    */
-  public void removeDiscrepancy(final AuditReason the_reason) {
-    ensureCounterExists(my_discrepancies, the_reason);
-    my_discrepancies.put(the_reason, my_discrepancies.get(the_reason) - 1);
-    if (my_current_round_index != null) {
-      my_rounds.get(my_current_round_index).removeDiscrepancy(the_reason);
+  public void removeDiscrepancy(final Set<AuditReason> the_reasons) {
+    final Set<AuditSelection> selections = new HashSet<>();
+    for (final AuditReason r : the_reasons) {
+      selections.add(r.selection());
     }
+    for (final AuditSelection s : selections) {
+      ensureCounterExists(my_discrepancies, s);
+      my_discrepancies.put(s, my_discrepancies.get(s) - 1);
```

Comment:
What if `my_discrepancies.get(s)` is 0, say, from the previous line?

Reply:
Also apply this comment to the `*Disagreement()` methods.

Reply:
Then it ends up negative. As it should.

Reply:
(I feel like we’ve had that particular discussion already :) )

Reply:
(Actually, that is only possible for round counters, not the overall dashboard counter, given the way those methods are used)